R/check_target_file_ext_valid.R

```diff
@@ -0,0 +1,42 @@
+#' Check that a target data file has a valid extension.
+#'
+#' Note that files which are part of a hive partitioned dataset must have
+#' parquet file extension only.
+#' @inheritParams check_target_file_name
+#' @inherit check_valid_round_id return
+#' @export
+check_target_file_ext_valid <- function(file_path) {
+  hive <- is_hive_partitioned_path(file_path, strict = FALSE)
+  if (hive) {
+    valid_ext <- c("parquet")
+  } else {
+    valid_ext <- c("csv", "parquet")
+  }
+  file_ext <- fs::path_ext(file_path)
+
+  check <- file_ext %in% valid_ext
+  details <- NULL
+
+  if (!check) {
+    invalid_ext <- setdiff(file_ext, valid_ext) # nolint: object_usage_linter
+    details <- cli::format_inline(
+      "Extension {.val {invalid_ext}} is not.
+    Must be one of {.val {valid_ext}}."
```

Comment:
Another suggestion: change `"Must be one of {.val {valid_ext}}."` to `"Must be {?/one of }{.val {valid_ext}}."`. This way, hive-partition errors will say "Must be parquet."

(not formatted as a suggestion because my previous suggestion encompasses this one.

Reply:
Fixed in c35e67abb7ac93f408e10924916b3dc9030adda9